code/modules/mob/living/carbon/xenomorph/damage_procs.dm

```diff
@@ -291,3 +291,26 @@
 
 	var/list/overlap = iff_tag.faction_groups & access_to_check
 	return length(overlap)
+
+/mob/living/carbon/Xenomorph/handle_flamer_fire(obj/flamer_fire/fire, var/damage, var/delta_time)
+	. = ..()
+	if(caste && (caste.fire_immunity & (FIRE_IMMUNITY_NO_DAMAGE|FIRE_IMMUNITY_NO_IGNITE)))
+		return
```

Comment:
See above, already automatically compensated for when type_b_debuff_xeno_armor() is called.

Reply:
I just rebased my branch to latest Master iteration and these two lines do not exist. Where are these from, exactly?